frontend/pkg/frontend/middleware_logging.go

```diff
@@ -81,11 +78,7 @@ func MiddlewareLoggingPostMux(w http.ResponseWriter, r *http.Request, next http.
 
 	correlationData := arm.NewCorrelationData(r)
 
-	// NOTE: Here the middleware span is extended by further attributes.
-	// If the tracingMiddleware is not registered, this lines will have no effect.
-	span := trace.SpanFromContext(ctx)
-	span.SetAttributes(attribute.String("correlation.id", correlationData.CorrelationRequestID))
-	span.SetAttributes(attribute.String("client.request.id", correlationData.ClientRequestID))
+	ctx = ContextWithTraceCorrelationData(ctx, correlationData)
```

Comment:
I debated whether to suggest moving the function definition to this file, because each middleware file was intended to be self-contained.  I assume it's elsewhere to keep all the OpenTelemetry logic in one place?

Can you at least add a comment here like
```
// ContextWithTraceCorrelationData is defined in middleware_tracing.go
```

Reply:
I see, y the thought was to keep the tracing peaces on one place. I will add a comment for now :)